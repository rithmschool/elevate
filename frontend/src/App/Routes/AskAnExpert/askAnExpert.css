.askAnExpert {
  text-align: center;
  margin: 4em;
}

.askAnExpert > p {
  margin: 2em;
}

.askAnExpert > h4 {
  margin: 2em;
}
