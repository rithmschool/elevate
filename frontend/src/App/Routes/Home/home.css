.Home_container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-family: 'Raleway', 'Lucida Sans';
  width: 100vw;
  height: 80vh;
}

.Home_container h1 {
  font-weight: bold;
  color: #6558f5;
}

.Home_container button {
  background-color: #6558f5;
  color: white;
  font-weight: bold;
  font-size: x-large;
  width: 25vw;
}

.Home_container button:hover{
  background:#31168B ;
  color: white;
}

.Home_template,
.Home_database {
  border: 1px #dcd9d5 solid;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 5px;
  width: 100vw;
  height: 50vh;
  text-align: center;
}

.Home_database {
  background-color: #dfe7ed;
}

.fa-user-tie,
.fa-chart-bar {
  padding: 3% 3%;
  border: 2px solid black;
  font-size: 5em;
  text-align: center;
  margin: 5%;
}

.Home_get-started {
  background-color: #dfe7ed;
  border: 1px #dcd9d5 solid;
  align-items: center;
  justify-content: center;
  margin-top: 5px;
  width: 100vw;
  height: 40vh;
  text-align: center;
  padding: 8% 0;
}

.Home_stats {
  border: 1px #dcd9d5 solid;
  position: relative;
  align-items: center;
  justify-content: center;
  margin-top: 5px;
  text-align: center;
}

.Home_circle {
  margin: 5%;
  display: inline-block;
  padding: 20px;
  width: 10em;
  height: 10em;
  border-radius: 50%;
  border: 1px solid black;
}

.Home_circle p {
  font-size: 1.5rem;
  line-height: 30px;
}

@media only screen and (max-width: 767px) {
  .Home_image1 {
    object-fit: cover;
    width: 100vw;
    height: 767px;
  }
  .Home_h1 {
    font-size: 6vw;
  }
}
