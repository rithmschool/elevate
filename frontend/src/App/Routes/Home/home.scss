$primaryBtn: #5821b0;
$secondaryBtn: #31168b;

$primaryBkg: #7e30fc;
$secondaryBkg: #ad7cfc;
$tertiaryBkg: #825dbd;

$primaryGrey: #dad7de;
$secondaryGrey: #9e90e6;
$tertiaryGrey: #b8aec9;
$quartieryGrey: #aaa6b2;

.section {
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;
  min-height: 75vh;

  h1 {
    text-align: center;
  }
  .btn {
    background-color: $primaryBtn;
    border-color: $primaryBtn;
    font-weight: bold;
    color: white;
    font-size: 1.25em;
    width: 250px;
    margin: 15px;
    &:hover {
      cursor: pointer;
      background-color: $secondaryBtn !important;
    }
  }

  .img-container {
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    align-items: center;
    margin: 0 auto;
    height: 100%;
    width: 75%;
  }

  .content-container {
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: center;
  }
}

.Home_hero {
  position: relative;
  font-family: "Raleway", "Lucida Sans";
  background: rgba($primaryBkg, 0.5); /* Fallback for old browsers */
  background-image: linear-gradient(190deg, #e2b0ff 10%, #9f44d3 100%);
  min-height: 93vh;

  /* For SVG Background - generated on svgbackgrounds.com */
  background-color: #e2b0ff;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='100%25' height='100%25' viewBox='0 0 1600 800'%3E%3Cg fill-opacity='0.20'%3E%3Cpolygon fill='%23d99dfc' points='800 100 0 200 0 800 1600 800 1600 200'/%3E%3Cpolygon fill='%23d08bf8' points='800 200 0 400 0 800 1600 800 1600 400'/%3E%3Cpolygon fill='%23c77bf3' points='800 300 0 600 0 800 1600 800 1600 600'/%3E%3Cpolygon fill='%23be6bed' points='1600 800 800 400 0 800'/%3E%3Cpolygon fill='%23b45de5' points='1280 800 800 500 320 800'/%3E%3Cpolygon fill='%23aa50dd' points='533.3 800 1066.7 800 800 600'/%3E%3Cpolygon fill='%239f44d3' points='684.1 800 914.3 800 800 700'/%3E%3C/g%3E%3C/svg%3E");
  background-attachment: fixed;
  background-size: cover;

  svg {
    position: absolute;
    bottom: 0;
    width: 100%;
    height: 100px;
  }

  .hero-content {
    display: flex;
    flex-flow: column nowrap;
    justify-content: center;
    align-items: center;

    h1 {
      font-weight: bold;
      font-size: 3.5rem;
      color: $primaryBtn;
    }

    p {
      color: white;
      text-align: center;
    }
  }
}

.Home_template {
  h1 {
    color: $primaryBtn;
  }
  p .color-emphasis {
    color: $primaryBtn;
  }
}

.Home_database {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  background-color: rgba($color: $secondaryBkg, $alpha: 0.05);
  min-height: 75vh;

  .title-container {
    margin-bottom: 2rem;
    h1 {
      color: $primaryBtn;
      text-align: center;
    }
  }

  .col-container {
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    margin: 1rem auto;
    padding: 1rem;
    min-height: 450px;
    // border: 1px solid red;

    .img-container {
      display: flex;
      flex-direction: column;
      justify-content: center;
      align-items: center;
      // border: 1px solid green;
      margin-bottom: 2rem;
      min-height: 212px;
      img {
        width: 75%;
      }
    }

    .content-container {
      display: flex;
      flex-flow: column nowrap;
      justify-content: flex-start;
      align-items: center;
      // border: 1px solid blue;
      padding: 0.5rem 0;
      min-height: 170px;
      h4 {
        color: $primaryBtn;
        text-align: center;
      }
    }
  }

  /* Matches "Continued Support" img size relative to other imgs in DB section*/
  .col-container:last-child img {
    width: 90%;
  }
}

.Home_get-started {
  text-align: center;
  padding: 3rem;
  min-height: 80vh;
  background: rgba($primaryBkg, 0.6); //Save as fallback

  /* For SVG Background - generated on svgbackgrounds.com */
  background-color: rgba($primaryBkg, 0.6);
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='400' height='400' viewBox='0 0 200 200'%3E%3Cg fill='none' stroke='%234919ff' stroke-width='1' stroke-opacity='0.2'%3E%3Crect x='-40' y='40' width='75' height='75'/%3E%3Crect x='-35' y='45' width='65' height='65'/%3E%3Crect x='-30' y='50' width='55' height='55'/%3E%3Crect x='-25' y='55' width='45' height='45'/%3E%3Crect x='-20' y='60' width='35' height='35'/%3E%3Crect x='-15' y='65' width='25' height='25'/%3E%3Crect x='-10' y='70' width='15' height='15'/%3E%3Crect x='-5' y='75' width='5' height='5'/%3E%3Crect width='35' height='35'/%3E%3Crect x='5' y='5' width='25' height='25'/%3E%3Crect x='10' y='10' width='15' height='15'/%3E%3Crect x='15' y='15' width='5' height='5'/%3E%3Crect x='40' width='75' height='75'/%3E%3Crect x='45' y='5' width='65' height='65'/%3E%3Crect x='50' y='10' width='55' height='55'/%3E%3Crect x='55' y='15' width='45' height='45'/%3E%3Crect x='60' y='20' width='35' height='35'/%3E%3Crect x='65' y='25' width='25' height='25'/%3E%3Crect x='70' y='30' width='15' height='15'/%3E%3Crect x='75' y='35' width='5' height='5'/%3E%3Crect x='40' y='80' width='35' height='35'/%3E%3Crect x='45' y='85' width='25' height='25'/%3E%3Crect x='50' y='90' width='15' height='15'/%3E%3Crect x='55' y='95' width='5' height='5'/%3E%3Crect x='120' y='-40' width='75' height='75'/%3E%3Crect x='125' y='-35' width='65' height='65'/%3E%3Crect x='130' y='-30' width='55' height='55'/%3E%3Crect x='135' y='-25' width='45' height='45'/%3E%3Crect x='140' y='-20' width='35' height='35'/%3E%3Crect x='145' y='-15' width='25' height='25'/%3E%3Crect x='150' y='-10' width='15' height='15'/%3E%3Crect x='155' y='-5' width='5' height='5'/%3E%3Crect x='120' y='40' width='35' height='35'/%3E%3Crect x='125' y='45' width='25' height='25'/%3E%3Crect x='130' y='50' width='15' height='15'/%3E%3Crect x='135' y='55' width='5' height='5'/%3E%3Crect y='120' width='75' height='75'/%3E%3Crect x='5' y='125' width='65' height='65'/%3E%3Crect x='10' y='130' width='55' height='55'/%3E%3Crect x='15' y='135' width='45' height='45'/%3E%3Crect x='20' y='140' width='35' height='35'/%3E%3Crect x='25' y='145' width='25' height='25'/%3E%3Crect x='30' y='150' width='15' height='15'/%3E%3Crect x='35' y='155' width='5' height='5'/%3E%3Crect x='200' y='120' width='75' height='75'/%3E%3Crect x='40' y='200' width='75' height='75'/%3E%3Crect x='80' y='80' width='75' height='75'/%3E%3Crect x='85' y='85' width='65' height='65'/%3E%3Crect x='90' y='90' width='55' height='55'/%3E%3Crect x='95' y='95' width='45' height='45'/%3E%3Crect x='100' y='100' width='35' height='35'/%3E%3Crect x='105' y='105' width='25' height='25'/%3E%3Crect x='110' y='110' width='15' height='15'/%3E%3Crect x='115' y='115' width='5' height='5'/%3E%3Crect x='80' y='160' width='35' height='35'/%3E%3Crect x='85' y='165' width='25' height='25'/%3E%3Crect x='90' y='170' width='15' height='15'/%3E%3Crect x='95' y='175' width='5' height='5'/%3E%3Crect x='120' y='160' width='75' height='75'/%3E%3Crect x='125' y='165' width='65' height='65'/%3E%3Crect x='130' y='170' width='55' height='55'/%3E%3Crect x='135' y='175' width='45' height='45'/%3E%3Crect x='140' y='180' width='35' height='35'/%3E%3Crect x='145' y='185' width='25' height='25'/%3E%3Crect x='150' y='190' width='15' height='15'/%3E%3Crect x='155' y='195' width='5' height='5'/%3E%3Crect x='160' y='40' width='75' height='75'/%3E%3Crect x='165' y='45' width='65' height='65'/%3E%3Crect x='170' y='50' width='55' height='55'/%3E%3Crect x='175' y='55' width='45' height='45'/%3E%3Crect x='180' y='60' width='35' height='35'/%3E%3Crect x='185' y='65' width='25' height='25'/%3E%3Crect x='190' y='70' width='15' height='15'/%3E%3Crect x='195' y='75' width='5' height='5'/%3E%3Crect x='160' y='120' width='35' height='35'/%3E%3Crect x='165' y='125' width='25' height='25'/%3E%3Crect x='170' y='130' width='15' height='15'/%3E%3Crect x='175' y='135' width='5' height='5'/%3E%3Crect x='200' y='200' width='35' height='35'/%3E%3Crect x='200' width='35' height='35'/%3E%3Crect y='200' width='35' height='35'/%3E%3C/g%3E%3C/svg%3E");

  .content-container.cta {
    background: white;
  }

  h2 {
    color: $secondaryBtn;
    font-weight: bold;
  }
}

.Home_about {
  background: rgba($secondaryGrey, 0.2);

  /* For SVG Background - generated on svgbackgrounds.com */
  background-color: rgba($secondaryGrey, 0.2);
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 1600 900'%3E%3Cpolygon fill='%23a176cc' points='957 450 539 900 1396 900'/%3E%3Cpolygon fill='%238025aa' points='957 450 872.9 900 1396 900'/%3E%3Cpolygon fill='%239c66c7' points='-60 900 398 662 816 900'/%3E%3Cpolygon fill='%23851e9d' points='337 900 398 662 816 900'/%3E%3Cpolygon fill='%239755c1' points='1203 546 1552 900 876 900'/%3E%3Cpolygon fill='%23881790' points='1203 546 1552 900 1162 900'/%3E%3Cpolygon fill='%239243ba' points='641 695 886 900 367 900'/%3E%3Cpolygon fill='%23891284' points='587 900 641 695 886 900'/%3E%3Cpolygon fill='%238f2cb2' points='1710 900 1401 632 1096 900'/%3E%3Cpolygon fill='%23890e79' points='1710 900 1401 632 1365 900'/%3E%3Cpolygon fill='%238b03aa' points='1210 900 971 687 725 900'/%3E%3Cpolygon fill='%23880c6e' points='943 900 1210 900 971 687'/%3E%3C/svg%3E");
  background-attachment: fixed;
  background-size: cover;
  background-position: center;

  h1 {
    color: $primaryBtn;
    text-align: center;
  }

  h4 {
    color: black;
    text-align: center;
  }

  p {
    color: black;
    text-align: justify;
  }

  .color-emphasis {
    color: $primaryBtn;
  }
}
