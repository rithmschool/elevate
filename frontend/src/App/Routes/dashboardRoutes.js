import React from "react";
import { Route, Redirect } from "react-router-dom";
import { UserContext } from "../../userContext";

/** if user isnt connected can't access to Dashboard routes */
class DashboardRoutes extends React.Component {
  static contextType = UserContext;
  render() {
    if (!localStorage.token) {
      return <Redirect to="/login" />;
    }

    return (
      <Route
        exact={this.props.exact}
        path={this.props.path}
        render={this.props.render}
      />
    );
  }
}

export default DashboardRoutes;
