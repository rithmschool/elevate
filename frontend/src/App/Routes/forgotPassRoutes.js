import React from "react";
import { Route, Redirect } from "react-router-dom";

/** if user is connected can't access to reseting password routes */
class ForgotPassRoutes extends React.Component {
  render() {
    const token = localStorage.getItem("token");

    if (token) return <Redirect to="/" />;

    return (
      <Route
        exact={this.props.exact}
        path={this.props.path}
        render={this.props.render}
      />
    );
  }
}

export default ForgotPassRoutes;
