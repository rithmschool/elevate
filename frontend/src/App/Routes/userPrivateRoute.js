import React, { Component } from "react";
import { Route, Redirect } from "react-router-dom";
import { UserContext } from "../../userContext";

class UserPrivateRoute extends Component {
  static contextType = UserContext;

  render() {
    // this.context will be the currentUser object or null on hard refresh

    if (!this.context || !this.context.userId) {
      return <Redirect to="/login" />;
    }

    return (
      <Route
        exact={ this.props.exact }
        path={ this.props.path }
        render={ this.props.render } />
    );
  }
}

export default UserPrivateRoute;
