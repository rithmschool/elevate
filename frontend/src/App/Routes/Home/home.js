import React from "react";
import { Link } from "react-router-dom";
import { Container, Row, Col, Image } from "react-bootstrap";

import "./home.scss";
import Footer from "./Footer/footer";

import file from "./img/files.png";
import person from "./img/person.png";
import completed from "./img/completed.png";
import fitting_pieces from "./img/undraw_fitting_pieces_cli3.svg";
import road_sign from "./img/undraw_road_sign_mfpo_alt.svg";
import stand_out from "./img/undraw_stand_out.svg";

class Home extends React.Component {
  render() {
    const routeStr = localStorage.token ? "/ask-an-expert" : "/login";
    return (
      <div>
        {/* Hero Section */}
        <Container fluid className="section Home_hero">
          <Row>
            <Col md={6} className="p-4">
              <div className="img-container">
                <Image fluid src={fitting_pieces} alt="fitting pieces" />
              </div>
            </Col>
            <Col md={6} className="content-container mb-3">
              <div className="hero-content">
                <h1>Your guide to your career</h1>
                <p className="lead my-3">
                  We all need help managing our careers. We got you covered.
                </p>
              </div>
              <Link to={routeStr} className="btn btn-primary">
                Get Started
              </Link>
            </Col>
          </Row>
        </Container>

        {/* SECTION - TEMPLATE - WHY BRELLA */}
        <Container fluid className="section Home_template p-5">
          <Row>
            <Col md={6} className="content-container mb-3">
              <div>
                <h1>Why Brella?</h1>
                <p className="my-4 lead">
                  Everything about your career is personal, so why isn’t your support?
                </p>
                <p className="lead">
                  We take the stress out of finding trustworthy, affordable advice by providing you
                  tools and experts to help you advance in your career. Wherever your career takes
                  you, your Brella team goes with you — we’re tied to you, not your employer.
                </p>
              </div>
            </Col>
            <Col md={6}>
              <div className="img-container">
                <Image fluid src={stand_out} alt="stand out" />
              </div>
            </Col>
          </Row>
        </Container>

        {/* SECTION - Database */}
        <Container fluid className="Home_database p-5">
          <Row>
            <Col>
              <h1 className="Home_database-title">The Brella Experience</h1>
            </Col>
          </Row>
          <Row>
            <Col md={4}>
              <div className="Home_database-img-container">
                <img src={file} alt="startFile" />
                <h4>Understand your offer letter</h4>
              </div>
              <div>
                <p className="Home_database-text lead">
                  Receive a report with an analysis of your offer, outstanding information, market
                  salary data, and next steps.
                </p>
              </div>
            </Col>
            <Col md={4}>
              <div className="Home_database-img-container">
                <img src={person} alt="person" />
                <h4>Take Action</h4>
              </div>
              <p className="Home_database-text lead">
                Partner with your Brella team to create a plan to help you get the offer you
                deserve.
              </p>
            </Col>
            <Col md={4}>
              <div className="Home_database-img-container">
                <img src={completed} alt="completed" />
                <h4>Continued support</h4>
              </div>
              <p className="Home_database-text lead">
                Work with our coaches to manage and grow your career and your compensation.
              </p>
            </Col>
          </Row>
        </Container>

        {/* SECTION - GET STARTED */}
        <Container fluid className="section Home_get-started">
          <Row className="p-5">
            <Col md={6} className="p-4">
              <div className="img-container">
                <Image fluid src={road_sign} alt="road sign" />
              </div>
            </Col>
            <Col md={6} className="content-container cta p-4 rounded-lg shadow-lg">
              <div>
                <h2>Unsure if you’re getting paid what you deserve?</h2>
                <p className="lead">
                  We believe you have a right to know. We’re on a mission to{" "}
                  <b>increase pay transparency</b> — and we need your help. Share your offer letter,
                  and we’ll share benchmark data with you as soon as we have it.
                </p>
              </div>
              <Link to={routeStr} className="btn btn-primary">
                Share your offer letter
              </Link>
            </Col>
          </Row>
        </Container>
        {/* FOOTER */}
        <Footer />
      </div>
    );
  }
}

export default Home;
