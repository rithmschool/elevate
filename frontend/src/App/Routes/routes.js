import React from "react";
import { Route, Switch, Redirect } from "react-router-dom";

import ForgotPassword from "./ForgotPassword/forgotPassword";
import ResetPassword from "./ResetPassword/resetPassword";
import Home from "./Home/home";
import UserProfile from "./UserProfile/userProfile";
import LoginSignUpForm from "./LoginSignUpForm/loginSignUpForm";
import AdminPanel from "./AdminPanel/adminPanel";
import AskAnExpert from "./AskAnExpert/askAnExpert";
import Dashboard from "./../Routes/Dashboard/dashboard";
import DashboardAppointments from "./../Routes/Dashboard/dashboardAppointments";
import DashboardManage from "./../Routes/Dashboard/dashboardManage";
import DashboardTemplates from "./../Routes/Dashboard/dashboardTemplates";

import AdminPrivateRoute from "./adminPrivateRoute";
import ForgotPassRoutes from "./forgotPassRoutes";
import UserPrivateRoute from "./userPrivateRoute";
import DashboardRoutes from "./dashboardRoutes";

class Routes extends React.Component {
  render() {
    return (
      <Switch>
        <Route exact path='/' render={props => <Home {...props} />} />

        <Route
          exact
          path='/login'
          render={props => (
            <LoginSignUpForm
              {...props}
              getCurrentUser={this.props.getCurrentUser}
            />
          )}
        />

        <AdminPrivateRoute
          exact
          path='/admin'
          render={props => <AdminPanel {...props} />}
        />

        <Route exact path='/ask-an-expert' render={() => <AskAnExpert />} />

        <UserPrivateRoute
          path='/users/:userId'
          render={props => <UserProfile {...props} />}
        />

        <DashboardRoutes exact path='/dashboard' render={() => <Dashboard />} />
        <DashboardRoutes
          exact
          path='/dashboard/manage'
          render={() => <DashboardManage />}
        />
        <DashboardRoutes
          exact
          path='/dashboard/appointments'
          render={() => <DashboardAppointments />}
        />
        <DashboardRoutes
          exact
          path='/dashboard/templates'
          render={() => <DashboardTemplates />}
        />

        <ForgotPassRoutes
          exact
          path='/reset-password/forgot'
          render={() => <ForgotPassword />}
        />

        <ForgotPassRoutes
          exact
          path='/reset-password/:token'
          render={props => <ResetPassword {...props} />}
        />

        <Redirect to='/' />
      </Switch>
    );
  }
}

export default Routes;
