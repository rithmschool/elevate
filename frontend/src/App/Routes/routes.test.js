import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Routes from "./routes";
import { MemoryRouter } from "react-router-dom";

import Home from "./Home/home";
import LoginSignupForm from "./LoginSignUpForm/loginSignUpForm";
import AskAnExpert from "./AskAnExpert/askAnExpert";
import App from "../app";

it("renders without crashing", function() {
  shallow(<Routes />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<Routes />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it("invalid path should redirect to /", function() {
  const wrapper = mount(
    <MemoryRouter initialEntries={["/invalidendpoint"]}>
      <App />
    </MemoryRouter>
  );
  expect(wrapper.find(Home)).toHaveLength(1);
});

it("should show Home component for / route (using memory router)", function() {
  const wrapper = mount(
    <MemoryRouter initialEntries={["/"]}>
      <Routes />
    </MemoryRouter>
  );
  expect(wrapper.find(Home)).toHaveLength(1);
});

it("should show LoginSignupForm component for /login route (using memory router)", function() {
  const wrapper = mount(
    <MemoryRouter initialEntries={["/login"]}>
      <Routes />
    </MemoryRouter>
  );
  expect(wrapper.find(LoginSignupForm)).toHaveLength(1);
});

it("should show AskAnExpert component for /ask-an-expert route (using memory router)", function() {
  const wrapper = mount(
    <MemoryRouter initialEntries={["/ask-an-expert"]}>
      <Routes />
    </MemoryRouter>
  );
  expect(wrapper.find(AskAnExpert)).toHaveLength(1);
});


