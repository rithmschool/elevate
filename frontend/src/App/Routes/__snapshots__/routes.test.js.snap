// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main routes file matches snapshot 1`] = `
<Switch>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/login"
    render={[Function]}
  />
  <AdminPrivateRoute
    exact={true}
    path="/admin"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/ask-an-expert"
    render={[Function]}
  />
  <UserPrivateRoute
    path="/users/:userId"
    render={[Function]}
  />
  <ForgotPassRoutes
    exact={true}
    path="/reset-password/forgot"
    render={[Function]}
  />
  <ForgotPassRoutes
    exact={true}
    path="/reset-password/:token"
    render={[Function]}
  />
  <Redirect
    to="/"
  />
</Switch>
`;
